doc/factors.md

```diff
@@ -0,0 +1,208 @@
+# Using marian with factors
```

Comment:
@graemenail Any suggestions how we should integrate this document into the new format of `doc/` and Sphinx-based docs? I'm thinking of creating a new section on the nav bar for user documentation and put it there.